### PR TITLE
RMTEAM-323: Add Rewriter for SetupResultFor

### PR DESCRIPTION
**Pull request by derlistigelurch**

### CODE REVIEW

File: Core/Rewriters/SetupResultForRewriter.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using ModelExtensions = Microsoft.CodeAnalysis.ModelExtensions;
+using SyntaxNodeExtensions = Microsoft.CodeAnalysis.SyntaxNodeExtensions;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters
+{
+  public class SetupResultForRewriter : RewriterBase
+  {
+    public override SyntaxNode? VisitExpressionStatement (ExpressionStatementSyntax node)
+    {
+      var trackedNodes = SyntaxNodeTrackingExtensions.TrackNodes (node, node.DescendantNodesAndSelf(), CompilationId);
```
Comment (iam-mholle) [inline Core/Rewriters/SetupResultForRewriter.cs]:
```suggestion
      var trackedNodes = node.TrackNodes (node.DescendantNodesAndSelf(), CompilationId);
```

Why so fancy?

File: Core/Rewriters/SetupResultForRewriter.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      var baseCallNode = (ExpressionStatementSyntax) base.VisitExpressionStatement (trackedNodes)!;
+
+      var setupResultSymbol = Model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.SetupResult");
+      if (setupResultSymbol == null)
+      {
+        throw new InvalidOperationException ("Moq cannot be found.");
```
Comment (iam-mholle) [inline Core/Rewriters/SetupResultForRewriter.cs]:
```suggestion
        throw new InvalidOperationException ("Rhino.Mocks cannot be found.");
```

File: Core/Rewriters/SetupResultForRewriter.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using CSharpExtensions = Microsoft.CodeAnalysis.CSharpExtensions;
+using ModelExtensions = Microsoft.CodeAnalysis.ModelExtensions;
+using SyntaxNodeExtensions = Microsoft.CodeAnalysis.SyntaxNodeExtensions;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters
+{
+  public class SetupResultForRewriter : RewriterBase
```
Comment (iam-mholle) [inline Core/Rewriters/SetupResultForRewriter.cs:22]: Why didn't you use the extensions methods as extensions methods in this file?

gif

File: Core/Rewriters/SetupResultForRewriter.cs (hunk 1 above)
Comment (iam-mholle) [inline Core/Rewriters/SetupResultForRewriter.cs]: Don't you only need to track expression statements here?

**Review by iam-mholle — CHANGES_REQUESTED** [review]

File: Core/Rewriters/SetupResultForRewriter.cs (hunk 1 above)
Comment (derlistigelurch) [reply Core/Rewriters/SetupResultForRewriter.cs]: > Don't you only need to track expression statements here?

No I also need to track the `SetupResult` and `For` identifiername and the easiest way to do this is to track every descendant node

Comment (iam-mholle) [reply Core/Rewriters/SetupResultForRewriter.cs]: > ... I also need to ...

Also? shouldn't you only need to track `SimpleMemberAccessExpression` nodes then?

**Review by iam-mholle — APPROVED** [review]

**Pull request merged**